packages/grid/x-data-grid/src/components/GridPagination.tsx

```diff
@@ -31,38 +31,41 @@ export const GridPagination = React.forwardRef<HTMLDivElement, Partial<TablePagi
     const paginationState = useGridSelector(apiRef, gridPaginationSelector);
 
     const lastPage = React.useMemo(
-      () => Math.floor(paginationState.rowCount / (paginationState.pageSize || 1)),
-      [paginationState.rowCount, paginationState.pageSize],
+      () => Math.floor(paginationState.rowCount / (paginationState.paginationModel.pageSize || 1)),
+      [paginationState.rowCount, paginationState.paginationModel],
```

Comment:
Do  not change when `page` is modified
```suggestion
      [paginationState.rowCount, paginationState.paginationModel.pageSize],
```